Sources/DefaultStringConvertible.swift

```diff
@@ -187,28 +187,42 @@ private func generateDeepDescription(any: Any) -> String {
         }
         return string
 
-    case .Struct, .Class:
+    case .struct, .class:
         if let any = any as? CustomDebugStringConvertible {
             return any.debugDescription
         }
 
-        var superclassMirror = mirror.superclassMirror()
+        var superclassMirror = mirror.superclassMirror
         repeat {
             if let superChildren = superclassMirror?.children {
-                properties.appendContentsOf(superChildren)
+                properties.append(contentsOf: superChildren)
             }
 
-            superclassMirror = superclassMirror?.superclassMirror()
+            superclassMirror = superclassMirror?.superclassMirror
         } while superclassMirror != nil
 
-        if properties.isEmpty { return "\(typeName)\(String(any))" }
+        if properties.isEmpty { return "\(typeName)\(String(describing: any))" }
         var string = "\(typeName){"
-        for (index, property) in properties.enumerate() {
+        for (index, property) in properties.enumerated() {
             string += indentedString("\(property.label!): \(generateDeepDescription(property.value))" + (index < properties.count - 1 ? ",\r" : ""))
         }
         return string + "\r}"
 
-    case .Optional:
+    case .optional:
         return generateDefaultDescription(any)
     }
 }
+
+// Since these methods are not available in Linux
+#if !(os(macOS) || os(iOS) || os(watchOS) || os(tvOS))
```

Comment:
We can, but this will decrease the level of generality.Swift sometime will come to platforms other than Linux and macOS.